crates/anvil/src/eth/api.rs

```diff
@@ -1312,6 +1318,61 @@ impl EthApi {
         .map(U256::from)
     }
 
+    /// Handler for RPC call: `anvil_getBlobByHash`
+    pub async fn anvil_get_blob_by_versioned_hash(
+        &self,
+        hash: B256,
+    ) -> Result<Option<alloy_consensus::TxEip4844WithSidecar>> {
+        node_info!("anvil_getBlobByHash");
+        let latest_block = self.backend.best_number();
+        for i in 0..=latest_block {
+            if let Some(block) = self.backend.get_full_block(i) {
+                for tx in block.into_transactions_iter() {
+                    let typed_tx = TypedTransaction::try_from(tx).unwrap();
+
+                    if let TypedTransaction::EIP4844(signed) = typed_tx
+                        && let Ok(sidecar_tx) = signed.tx().clone().try_into_4844_with_sidecar()
+                    {
+                        for item in sidecar_tx.sidecar.clone() {
+                            let versioned_hash = B256::from(item.to_kzg_versioned_hash());
+
+                            if versioned_hash == hash {
+                                return Ok(Some(sidecar_tx));
+                            }
+                        }
+                    }
+                }
```

Comment:
this is a bit expensive, what we can do instead is replicate this fn on Backend and so that this can iterate over `self.blockchain.storage.read()` without cloning everything, like

let storage = self.blockchain.storage.read()
for block in storage.blocks {
}